query/src/servers/clickhouse/interactive_worker.rs

```diff
@@ -113,7 +110,7 @@ impl ClickHouseSession for InteractiveWorker {
                 .await
             {
                 Ok(user_info) => (
-                    user_manager.auth_user(user_info.clone(), info).await,
+                    user_info.auth_info.auth_password_plaintext(password),
```

Comment:
if the password is sha256, `auth_password_plaintext` does not work?

Reply:
Looks we need a `auth_info.auth(type:ClickHouse/MySQL, Option<salt>, password)` to do all the handlers type auth?
clickhouse also has the SHA256 password type.

Reply:
both clickhouse and MySQL support it，and store sha2hashed value，but the protocol parts are different.

Reply:
it seems clickhouse client transport password in plaintext directly without salt，so this interface can be used by ch and http basic

Reply:
only mysql client implement the protocol， others just share the password stored

Reply:
> Looks we need a auth_info.auth(type:ClickHouse/MySQL, Option<salt>, password) to do all the handlers type auth?

+1, a unified `auth()` would be easier to reason about? q.q

Reply:
@flaneur2020  thanks for review

double_sha by its name only means how the password is stored. if the password is safely delivered, we valid it by simply hashing the input again, that is what the clickhouse-client did, and we will deal with it the same way for http-basic. 

but for MySQL client/diver it also means an auth protocol, the validation method is totally different too.

for stored password auth only, I`d rather sep them apart.

for the general auth interface, This pr is not the final interface,  it needs further refactor when external auth is introduced. we need at least an auth manager object because each type of external auth may have multi instances with diff configs to load, and the unified auth entrance should be here. I haven't quite nailed that down yet, but I think the usage of the enum AuthInfo is a good start,  which is the main point for the pr.